### PR TITLE
Updates to CMakelists.txt to export target and create pkg-config file, adding meson build file

### PR DESCRIPTION
**Pull request by telegraphic**
**Description of pull request**

This PR is an update to `CMakelists.txt` to make it easier to use astro-accelerate as a dependency.

When `make install` is run:
* An `astro-accelerateConfig.cmake` file for `find_package` will be installed to `${CMAKE_INSTALL_PREFIX}/cmake`.
* A pkg-config file is generated and installed to `${CMAKE_INSTALL_PREFIX}/lib/pkgconfig`.
* Header files are copied to `${CMAKE_INSTALL_PREFIX}/include/aa`
* `astroaccelerate` binary is copied to  `${CMAKE_INSTALL_PREFIX}/bin`

I also added some lines to quash cmake policy warnings that were popping up.

**Interfaces**
Will this pull request result in a backwards-incompatible interface change: no

Expected semantic version number increment category (Please indicate x.y.z): 1.3.2

**Notes**

This PR was motivated by my plan to make some utilities (plugins for bifrost) using astro-accelerate. Thought I should to share in case others could benefit. I want to use meson as a build tool -- I don't really know what I'm doing with cmake so my changes may not be the canonical approach!

### CODE REVIEW

**Comment by telegraphic** [comment]
Update: I have now added a `meson.build` file, so the project can be built with meson as well

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+comp_flags = ['-g', '-std=c++14']
+add_project_arguments(comp_flags, language : ['c', 'cpp', 'cuda'])
+
+
+# Load list of files to compile into shared object
+# Note: Meson does not allow * wildcard (deliberately)
+# Instead, we use file generated from command: ls src/*.cpp > meson_srclist.txt
+fs = import('fs') 
+srcs = fs.read('meson_srclist.txt').strip().split('\n')
+apps = fs.read('meson_applist.txt').strip().split('\n')
```
Comment (eli-schwartz) [inline meson.build]: File seems to be missing?

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Meson build script for pacer imager
+project('astroaccelerate', 'cpp', 'cuda', version: '1.3.1')
+
+# Basic setup
+cc = meson.get_compiler('cpp')
+c_std = 'c++14'
+buildtype = 'debug'
+comp_flags = ['-g', '-std=c++14']
```
Comment (eli-schwartz) [inline meson.build:10]: This should be `default_options: ['debug=true', 'cpp_std=c++14']` as a kwarg to the `project()` function.

File: astroaccelerate.pc.in [hunk 3]
```diff
@@ -0,0 +1,10 @@
+prefix=@DEST_DIR@
+libdir=${prefix}/lib
+includedir=${prefix}/include
```
Comment (eli-schwartz) [inline astroaccelerate.pc.in:3]: See https://github.com/jtojnar/cmake-snips/

This is a bit complicated to do correctly with cmake -- I strongly advise to use that snippet to construct paths.

Your meson.build contribution could just use
```meson
pkgconfig = import('pkgconfig')
pkgconfig.generate(
    astroac_lib,
    description: 'AstroAccelerate is a many-core accelerated software package for processing time-domain radio-astronomy data.',
)
```

and it should work out of the box.

Comment (telegraphic) [reply astroaccelerate.pc.in:3]: Thanks, this is indeed much nicer than cmake!

File: meson.build (hunk 2 above)
Comment (telegraphic) [reply meson.build:10]: Not sure why, but the flags aren't propagated to `nvcc` if I use this approach. (I'm using meson 0.62.0, and and cuda 11.5)

File: meson.build (hunk 1 above)
Comment (telegraphic) [reply meson.build]: Thanks, line now removed as the apps are listed in the `examples/meson.build`

**Comment by telegraphic** [comment]
thanks again @eli-schwartz!

File: meson.build (hunk 2 above)
Comment (eli-schwartz) [reply meson.build:10]: Well, if you leave it as it is now, you need to pass it to the `language: ['cpp', 'cuda']`.

So try passing `'cuda_std=c++14'` as well. :)

**Comment by telegraphic** [comment]
TODO: Add generation of `aa_version.hpp` via Meson.

**Comment by telegraphic** [comment]
DONE: need to change
```
-join_paths(lib\pkgconfig "\${prefix}" "${CMAKE_INSTALL_LIBDIR}")
+join_paths(CMAKE_INSTALL_LIBDIR "lib/pkgconfig" "${prefix}")
```

and need to add 
```
set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
include(JoinPaths)
```

**Comment by KAdamek** [comment]
Thanks for updating Cmake.

**Pull request merged**